src/model_code.py

```diff
@@ -96,6 +97,202 @@ def running_mean(x):
+    sigma2 = res.mse_total
+    vbeta = np.linalg.inv(X.T.dot(X))
+    betas[:, 0] = beta_hat.copy()
+    sigmas2[0] = sigma2
+
+    accepts_alpha0, accepts_alpha1, accepts_gamma0, accepts_gamma1 = 0, 0, 0, 0
+    ##################################################
+
+    ############ Sampling Gibbs + MH #################
+    #proposal_alpha0 = norm(loc=0, scale=taus[0])
+    #proposal_alpha1 = norm(loc=0, scale=taus[1])
+    #proposal_gamma0 = norm(loc=0, scale=taus[2])
+    #proposal_gamma1 = norm(loc=0, scale=taus[3])
+    proposal_alpha0 = norm(loc=alpha0, scale=taus[0])
+    proposal_alpha1 = norm(loc=alpha1, scale=taus[1])
+    #proposal_gamma0 = norm(loc=gamma0, scale=taus[2])
+    #proposal_gamma1 = norm(loc=gamma1, scale=taus[3])
+
+    for i in tqdm(range(1, N)):
+        # sample a beta
+        beta = mvnorm(mean=beta_hat, cov=sigmas2[i-1] * vbeta, allow_singular=True).rvs()
+        # sample a sigma2
+        sigma2 = invgamma.rvs(n / 2, scale=(y - X.dot(betas[:, i-1])).T.dot(y - X.dot(betas[:, i-1])) / 2)
+        # sample the missing higher_yes
+        p = np.exp(alpha0 + alpha1 * age_missing_higher_yes) / (1 + np.exp(alpha0 + alpha1 * age_missing_higher_yes))
+        higher_yes = bernoulli.rvs(p)
+        # sample the missing absences
+        mu = np.exp(gamma0 + gamma1 * age_missing_absences)
+        absences = poisson.rvs(mu)
+
+        # sample alpha0, alpha1
+        #proposal_alpha0 = norm(loc=alpha0, scale=taus[0])
+        alpha0_star = proposal_alpha0.rvs()
+        log_u = np.log(uniform.rvs())
+        log_r = (
+                dist_log_alpha0(alpha0_star, alpha1, higher_yes_sim[:, i-1], age_missing_higher_yes, loc, scale) -
+                dist_log_alpha0(alpha0, alpha1, higher_yes_sim[:, i-1], age_missing_higher_yes, loc, scale)
+        )
+        if log_u < log_r:
+            alpha0 = alpha0_star
+            accepts_alpha0 += 1
+
+        #proposal_alpha1 = norm(loc=alpha1, scale=taus[1])
+        alpha1_star = proposal_alpha1.rvs()
+        log_u = np.log(uniform.rvs())
+        log_r = (
+                dist_log_alpha1(alpha1_star, alphas0[i-1], higher_yes_sim[:, i-1], age_missing_higher_yes, loc, scale) -
+                dist_log_alpha1(alpha1, alphas0[i-1], higher_yes_sim[:, i-1], age_missing_higher_yes, loc, scale)
+        )
+        if log_u < log_r:
+            alpha1 = alpha1_star
+            accepts_alpha1 += 1
+
+        # sample gamma0, gamma1
+        proposal_gamma0 = norm(loc=gamma0, scale=taus[2])
+        gamma0_star = proposal_gamma0.rvs()
+        log_u = np.log(uniform.rvs())
+        log_r = (
+                dist_log_gamma0(gamma0_star, gamma1, absences_sim[:, i-1], age_missing_absences, loc, scale) -
+                dist_log_gamma0(gamma0, gamma1, absences_sim[:, i-1], age_missing_absences, loc, scale)
+        )
+        if log_u < log_r:
+            gamma0 = gamma0_star
+            accepts_gamma0 += 1
+
+        proposal_gamma1 = norm(loc=gamma1, scale=taus[3])
```

Comment:
I think we should handle the gammas the same way we do the alphas

Reply:
I pushed with updates